src/jbmc/jbmc_parse_options.cpp

```diff
@@ -508,15 +507,72 @@ int jbmc_parse_optionst::doit()
         });
     };
   }
+
+  if(!cmdline.isset("symex-driven-lazy-loading"))
+  {
+    std::unique_ptr<goto_modelt> goto_model_ptr;
+    int get_goto_program_ret=get_goto_program(goto_model_ptr, options);
+    if(get_goto_program_ret!=-1)
+      return get_goto_program_ret;
+
+    goto_modelt &goto_model = *goto_model_ptr;
+
+    if(cmdline.isset("show-properties"))
+    {
+      show_properties(
+        goto_model, get_message_handler(), ui_message_handler.get_ui());
+      return 0; // should contemplate EX_OK from sysexits.h
+    }
+
+    if(set_properties(goto_model))
+      return 7; // should contemplate EX_USAGE from sysexits.h
+
+    return bmct::do_language_agnostic_bmc(
+      options, goto_model, ui_message_handler.get_ui(), *this, configure_bmc);
+  }
   else
   {
-    configure_bmc = [](
-      bmct &bmc, const symbol_tablet &symbol_table) { // NOLINT (*)
-      // NOOP
+    // Use symex-driven lazy loading:
+    lazy_goto_modelt lazy_goto_model=lazy_goto_modelt::from_handler_object(
+      *this, options, get_message_handler());
+    lazy_goto_model.initialize(cmdline);
+
+    // The precise wording of this error matches goto-symex's complaint when no
+    // __CPROVER_start exists (if we just go ahead and run it anyway it will
+    // trip an invariant when it tries to load it)
+    if(!lazy_goto_model.symbol_table.has_symbol(goto_functionst::entry_point()))
+    {
+      error() << "the program has no entry point";
+      return 6;
+    }
+
+    // Add failed symbols for any symbol created prior to loading any
+    // particular function:
+    add_failed_symbols(lazy_goto_model.symbol_table);
+
+    // If applicable, parse the coverage instrumentation configuration, which
+    // will be used in process_goto_function:
+    cover_config =
+      get_cover_config(
+        options, lazy_goto_model.symbol_table, get_message_handler());
+
+    // Provide show-goto-functions and similar dump functions after symex
+    // executes. If --paths is active, these dump routines run after every
+    // paths iteration. Its return value indicates that if we ran any dump
+    // function, then we should skip the actual solver phase.
+    auto callback_after_symex = [this, &lazy_goto_model]() { // NOLINT (*)
+      return show_loaded_functions(lazy_goto_model);
     };
+
+    return
+      bmct::do_language_agnostic_bmc(
+        options,
+        lazy_goto_model,
+        ui_message_handler.get_ui(),
+        *this,
+        configure_bmc,
```

Comment:
Is this always null here? No - when `java-unwind-enum-static` is set? Perhaps a comment (though I'm wary it risks going out of date).

Reply:
Comment added